### PR TITLE
Bump CAPI to v1.6 and export CAPM3 diagnostics var for clusterctl

### PR DESCRIPTION
**Pull request by kashifest**
As the title says

### CODE REVIEW

File: tests/roles/run_tests/templates/clusterctl-vars.yaml [hunk 1]
```diff
@@ -26,3 +26,7 @@ EXTERNAL_SUBNET_V4_PREFIX: {{ EXTERNAL_SUBNET_V4_PREFIX }}
 EXTERNAL_SUBNET_V4_HOST: {{ EXTERNAL_SUBNET_V4_HOST }}
 EXTERNAL_SUBNET_V6_PREFIX: {{ EXTERNAL_SUBNET_V6_PREFIX }}
 EXTERNAL_SUBNET_V6_HOST: {{ EXTERNAL_SUBNET_V6_HOST }}
+
+# Metrics
+CAPM3_DIAGNOSTICS_ADDRESS: {{ CAPM3_DIAGNOSTICS_ADDRESS }}
+CAPM3_INSECURE_DIAGNOSTICS: {{ CAPM3_INSECURE_DIAGNOSTICS }}
```
Comment (tuminoid) [inline tests/roles/run_tests/templates/clusterctl-vars.yaml:32]: GH shows it is missing linefeed at the end.

File: tests/roles/run_tests/vars/main.yml [hunk 2]
```diff
@@ -87,6 +87,10 @@ IRONIC_TLS_SETUP: "{{ lookup('env', 'IRONIC_TLS_SETUP') | default('true', true)
 IRONIC_BASIC_AUTH: "{{ lookup('env', 'IRONIC_BASIC_AUTH') | default('true', true) }}"
 IRONIC_USE_MARIADB: "{{ lookup('env', 'IRONIC_USE_MARIADB') | default('false', true) }}"
 
+# Metrics
+CAPM3_DIAGNOSTICS_ADDRESS: "localhost:8080"
```
Comment (tuminoid) [inline tests/roles/run_tests/vars/main.yml:91]: Are these related to CAPI bump (other than helping debug things?)

Comment (kashifest) [reply tests/roles/run_tests/vars/main.yml:91]: Related to CAPI BUMP yes, CAPI has enabled this from CAPI v1.6 and we are following up with the same in CAPM3.

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Review by Rozzii — APPROVED** [review]
/lgtm

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Comment by kashifest** [comment]
/override test-centos-integration-release-1-5 
/override test-ubuntu-e2e-integration-main 
/hold 
Giving some more time to review.

**Pull request merged**